core/src/main/java/org/infinispan/configuration/parsing/Parser.java

```diff
@@ -1602,7 +1602,29 @@ private void parseDataContainer(final XMLExtendedStreamReader reader) throws XML
 
    private void parseMemory(final XMLExtendedStreamReader reader, final ConfigurationBuilderHolder holder) throws XMLStreamException {
       MemoryConfigurationBuilder memoryBuilder = holder.getCurrentConfigurationBuilder().memory();
-      ParseUtils.requireNoAttributes(reader);
+      if (reader.getSchema().since(11, 0)) {
```

Comment:
I am currently validating this conflict at builder time, to avoid putting logic in the XML parser since the JSON conversion doesn't see it.

Reply:
I believe it Is also logging deprecations in the builders, I will double check

Reply:
> I would say lets put the pre 11 schema reading in the if 

The parser is supposed to read XMLs with old schemas, doing this makes it incapable of parsing older files. This can be seen in a bunch of test failures in core

Reply:
Maybe I was unclear. I was more saying the code that is only for pre 11 should be in an if. Other code for 11+ would be inline with the method. Was all I was saying.

Reply:
There is not code that is pre 11 only. ISPN 11 will be able to parse both configurations, and throw an error if they are mixed. Or do you suggest to only support the new one?

Reply:
I don't quite remember what the position is for XML parsing @tristantarrant if we should just remove it in the major to begin with?

And regarding the if, I was more just saying we could put the code that is pre 11 in the if, so we can easily just delete the lines later, but that is just a nitpick.

Reply:
I am assuming we should be able to parse old and new files, to config eviction with the "old" and "new" attributes, it's been painful :)